### PR TITLE
gh-103629: Update typing.Unpack docs in compliance with PEP 692

### PR DESCRIPTION
**Pull request by franekmagiera**
* Issue: gh-103629

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Thanks! Could you add an entry to the 3.12 What's New and to the NEWS as well?

**Comment by franekmagiera** [comment]
Thought about doing it in a separate PR but I can add it as a part of this one :) let me do that.

**Review by AlexWaygood — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks @franekmagiera!

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -1428,6 +1433,21 @@ These are not used in annotations. They are building blocks for creating generic
       tup: tuple[*Ts]         # Syntax error on Python <= 3.10!
       tup: tuple[Unpack[Ts]]  # Semantically equivalent, and backwards-compatible
 
+   ``Unpack`` can also be used along with :class:`typing.TypedDict` for typing
+   ``**kwargs`` in a function signature::
+
+      from typing import TypedDict, Unpack
+
+      class Movie(TypedDict):
```
Comment (rinarakaki) [inline Doc/library/typing.rst:1441]: 4 space tab would've been more consistent.

Comment (JelleZijlstra) [reply Doc/library/typing.rst:1441]: 3 spaces are the convention in the docs.

**Pull request merged**